packages/xdg_directories/lib/xdg_directories.dart

```diff
@@ -131,14 +138,14 @@ List<Directory> get dataDirs {
 ///
 /// Throws [StateError] if the HOME environment variable is not set.
 Directory get dataHome =>
-    _directoryFromEnvironment('XDG_DATA_HOME', '.local/share');
+    _directoryFromEnvironmentWithFallback('XDG_DATA_HOME', '.local/share');
 
 /// The base directory relative to which user-specific runtime
 /// files and other file objects should be placed. (Corresponds to
 /// `$XDG_RUNTIME_DIR`).
 ///
 /// Throws [StateError] if the HOME environment variable is not set.
-Directory get runtimeDir => _directoryFromEnvironment('XDG_RUNTIME_DIR', null);
+Directory? get runtimeDir => _directoryFromEnvironment('XDG_RUNTIME_DIR');
 
 /// Gets the xdg user directory named by `dirName`.
 ///
```

Comment:
Based on a bug that came in recently, we should make `getUserDirectory` return `Directory?`, because I think we'll want to add logic to make it catch run failures (e.g., due to `xdg-user-dir` not being installed) and return null, and changing the return type later would be a breaking change.